apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/RetryOptions.cs

```diff
@@ -0,0 +1,56 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+namespace Google.Cloud.Storage.V1;
+
+/// <summary>
+/// Options used to indicate the retry configuration for an API request in case of failure. This class is immutable.
+/// These determine whether and how the request should be retried.
+/// </summary>
+public sealed class RetryOptions
+{
+    /// <summary>
+    /// Options used by default for idempotent operations upon failure.
+    /// It has all retry timings reset to defaults and retry predicate based on default error codes.
+    /// </summary>
+    public static RetryOptions IdempotentRetryOptions { get; } = new RetryOptions(RetryTiming.Default, RetryPredicate.RetriableIdempotentErrors);
+
+    /// <summary>
+    /// This configuration ensures that the API request is never retried on failure.
+    /// </summary>
+    public static RetryOptions Never { get; } = new RetryOptions(RetryTiming.Default, RetryPredicate.Never);
+
+    internal RetryPredicate Predicate { get; }
+    internal RetryTiming Timing { get; }
+
+    /// <summary>
+    /// In case of partially idempotent operations, this method checks the conditions required to mark it as retriable.
+    /// If the condition is available, RetryOptions are set to default idempotent values. If not, they are set to Never.
+    /// </summary>
+    /// <param name="condition">This condition's presence determines whether the operation is retried or not</param>
```

Comment:
I don't understand this as a description - in particular, there's no mention of null, or what kind of value this is otherwise likely to be. (If we didn't touch this code for 5 years and came back to it having forgotten how it's used, I don't think we'd understand this.)